### ISSUE TITLE
InternalServerError: 500 status code (no body) for Gemma 2 with system prompt

### ISSUE THREAD

**Issue by Qualzz** [EricLBuehler/mistral.rs#506]
I can't seem to be able to send an array of message to the openAI chat completion. 

```
const result = await openai.chat.completions.create({
        model: "gemma-2-9b-it",
        messages: [
            {"role": "user", "content": "You are Mistral.rs, an AI assistant."},
          ],
        max_tokens:2048,
        frequency_penalty:1.0,
        top_p:0.1,
        temperature:0,
      });
```

This works, but the following messages doesn't: 

```
messages: [
            {"role": "system", "content": "You are Mistral.rs, an AI assistant."},
            {"role": "user", "content": "Hello !!."},
        ],
```

Weirdly enough: 
```
messages: [
            {"role": "system", "content": "You are Mistral.rs, an AI assistant."},
         ],
```

this doesn't work, while replacing system by user works. 

TLDR: 

user > works
system, user > don't works
user, assistant > works
user, user > don't works

Maybe i'm doing something wrong ?

**Comment by EricLBuehler** [comment]
Hi @Qualzz! The Gemma 2 models do not support the system role - and we prioritize correctness, so this is handled as an error. 

Just to confirm, do you get any output, or does it only display the 500 Internal Server Error? Can you please paste the output for one of the failing messages?

**Comment by Qualzz** [comment]
Hello, the openAI node js module doesn't show the error, however I just did the same request in curl and got: 
`"Conversation roles must alternate user/assistant/user/assistant/"`
Which kinda fit my observations. 
Is there any workaround ? I was trying to create a discord bot, and thought that I could add a user:content for each discord message but it seems i'll need to find another way.

**Comment by EricLBuehler** [comment]
> Hello, the openAI node js module doesn't show the error

Hmm. When I run these commands in the Python server examples they seem to display the error.

> Is there any workaround ? I was trying to create a discord bot, and thought that I could add a user:content for each discord message but it seems i'll need to find another way.

With the incorrect alternating roles, that is a bit more tricky. It depends on what type of bot you are creating, if it needs to take all the messages, I would suggest concatenating them and perhaps encoding the usernames:
```
USER1: <text>
USER2: <text>
```
And then this would be fed into the `user` role.

**Comment by Qualzz** [comment]
I confirm that the openAi node js doesn't display the error, just a generic `InternalServerError: 500 status code (no body)`

I'll try ! You can close this issue if you think it should be closed now ! 
Thank you !

**Comment by EricLBuehler** [comment]
Ok, that sounds good. Please feel free to open a new issue if you have any questions or contact me on Discord, I'm excited to see how this develops.
